analysis/src/lib.rs

```diff
@@ -1,9 +1,33 @@
-use std::path::Path;
+mod xml;
 
-pub struct Analysis {}
+use std::{fs, path::Path};
+
+#[derive(Debug)]
+pub struct Analysis {
+    pub vk: Library,
+    pub video: Library,
+}
 
 impl Analysis {
-    pub fn new(_vulkan_headers_path: impl AsRef<Path>) -> Analysis {
-        Analysis {}
+    pub fn new(vulkan_headers_path: impl AsRef<Path>) -> Analysis {
+        let vulkan_headers_path = vulkan_headers_path.as_ref();
+        Analysis {
+            vk: Library::new(vulkan_headers_path.join("registry/vk.xml")),
+            video: Library::new(vulkan_headers_path.join("registry/video.xml")),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct Library {
+    _xml: xml::Registry,
+}
+
+impl Library {
+    fn new(xml_path: impl AsRef<Path>) -> Library {
+        let xml_input = Box::leak(fs::read_to_string(xml_path).unwrap().into_boxed_str());
```

Comment:
Why leak this, for roxml lifetimes? Should we clean it up in `Drop`, or is this intentionally skipped to keep all the string names `'static` (in which case that detail should be documented here)?

Reply:
I think it's worthwhile too, but let's document that for future readers.

Reply:
(In the current generator, wherever I've been able to clean up `String` copies, I just ended up threading through an explicit lifetime. Little less convenient to write, but skips the lifetime issues)